### PR TITLE
Dev secondary lr

### PR DESCRIPTION
**Pull request by simonJJJ**
为maskrcnn的secondary_lr需求，重新在sgd, momentum optim中支持secondary_lr，在PiecewiseScalingScheduler添加接口。

### CODE REVIEW

**Review by liujuncheng — CHANGES_REQUESTED** [review]
secondary_lr_lbn以及model_name是计划废弃的特性。
这里需要更详细描述下使用场景，以及其他框架如何解决这个问题的，然后我们设计更通用的方案。

**Comment by simonJJJ** [comment]
> secondary_lr_lbn以及model_name是计划废弃的特性。
> 这里需要更详细描述下使用场景，以及其他框架如何解决这个问题的，然后我们设计更通用的方案。

这里的使用场景是maskrcnn的bias learning rate和其他blob不同，pytorch支持这个非常简单，可以做到每个blob都有一个不同的lr, 只需要将参数和lr以字典的形式传入optimizer即可：
```
params = []
    for key, value in model.named_parameters():
        if not value.requires_grad:
            continue
        lr = cfg.SOLVER.BASE_LR
        weight_decay = cfg.SOLVER.WEIGHT_DECAY
        if "bias" in key:
            lr = cfg.SOLVER.BASE_LR * cfg.SOLVER.BIAS_LR_FACTOR
            weight_decay = cfg.SOLVER.WEIGHT_DECAY_BIAS
        params += [{"params": [value], "lr": lr, "weight_decay": weight_decay}]

optimizer = torch.optim.SGD(params, lr, momentum=cfg.SOLVER.MOMENTUM)```

**Comment by liujuncheng** [comment]
> > secondary_lr_lbn以及model_name是计划废弃的特性。
> > 这里需要更详细描述下使用场景，以及其他框架如何解决这个问题的，然后我们设计更通用的方案。
> 
> 这里的使用场景是maskrcnn的bias learning rate和其他blob不同，pytorch支持这个非常简单，可以做到每个blob都有一个不同的lr, 只需要将参数和lr以字典加列表的形式传入optimizer即可：
> 
> ```
> params = []
>     for key, value in model.named_parameters():
>         if not value.requires_grad:
>             continue
>         lr = cfg.SOLVER.BASE_LR
>         weight_decay = cfg.SOLVER.WEIGHT_DECAY
>         if "bias" in key:
>             lr = cfg.SOLVER.BASE_LR * cfg.SOLVER.BIAS_LR_FACTOR
>             weight_decay = cfg.SOLVER.WEIGHT_DECAY_BIAS
>         params += [{"params": [value], "lr": lr, "weight_decay": weight_decay}]
> 
> optimizer = torch.optim.SGD(params, lr, momentum=cfg.SOLVER.MOMENTUM)```
> ```

这是一个不同的parameter需要不同的超参（甚至有些网络parameter需要不同的优化方法）的问题，需要从这个角度解决。

**Comment by ScXfjiang** [comment]
> 这是一个不同的parameter需要不同的超参（甚至有些网络parameter需要不同的优化方法）的问题，需要从这个角度解决。

好的，我和shijie思考一下这里，看看其他框架是如何解决的，然后再找大家讨论。

**Comment by ScXfjiang** [comment]
https://github.com/Oneflow-Inc/oneflow/pull/4239
目前已经支持secondary learning rate，配置两个Optimizer即可，见test_multi_optimizer.py。
